statsmodels/tsa/exponential_smoothing/ets.py

```diff
@@ -0,0 +1,1679 @@
+            else:
+                raise ValueError(
+                    "Argument random_state must be None, an integer, "
+                    "or an instance of np.random.RandomState"
+                )
+        elif isinstance(random_errors, (rv_continuous, rv_discrete)):
+            params = random_errors.fit(self.resid)
+            eps = random_errors.rvs(*params, size=(nsimulations, repetitions))
+        elif isinstance(random_errors, _distn_infrastructure.rv_frozen):
+            eps = random_errors.rvs(size=(nsimulations, repetitions))
+        else:
+            raise ValueError("Argument random_errors has unexpected value!")
+
+        # get model settings
+        mul_seasonal = self.seasonal == "mul"
+        mul_trend = self.trend == "mul"
+        mul_error = self.error == "mul"
+
+        # define trend, damping and seasonality operations
+        if mul_trend:
+            op_b = np.multiply
+            op_d = np.power
+        else:
+            op_b = np.add
+            op_d = np.multiply
+        if mul_seasonal:
+            op_s = np.multiply
+        else:
+            op_s = np.add
+
+        # x translation:
+        # - x[t, 0, :] is level[t]
+        # - x[t, 1, :] is trend[t]
+        # - x[t, 2, :] is seasonal[t]
+        # - x[t, 3, :] is seasonal[t-1]
+        # - x[t, 2+j, :] is seasonal[t-j]
+        # - similarly: x[t-1, 2+m-1, :] is seasonal[t-m]
+        for t in range(nsimulations):
+            B = op_d(x[t - 1, 1, :], phi)
+            L = op_b(x[t - 1, 0, :], B)
+            S = x[t - 1, 2 + m - 1, :]
+            Y = op_s(L, S)
+            if self.error == "add":
+                eta = 1
+                kappa_l = 1 / S if mul_seasonal else 1
+                kappa_b = kappa_l / x[t - 1, 0, :] if mul_trend else kappa_l
+                kappa_s = 1 / L if mul_seasonal else 1
+            else:
+                eta = Y
+                kappa_l = 0 if mul_seasonal else S
+                kappa_b = (
+                    kappa_l / x[t - 1, 0, :]
+                    if mul_trend
+                    else kappa_l + x[t - 1, 0, :]
+                )
+                kappa_s = 0 if mul_seasonal else L
+
+            y[t, :] = Y + eta * eps[t, :]
+            x[t, 0, :] = L + alpha * (mul_error * L + kappa_l) * eps[t, :]
+            x[t, 1, :] = B + beta * (mul_error * B + kappa_b) * eps[t, :]
+            x[t, 2, :] = S + gamma * (mul_error * S + kappa_s) * eps[t, :]
+            # update seasonals by shifting previous seasonal right
+            x[t, 3:, :] = x[t - 1, 2:-1, :]
+
+        # Wrap data / squeeze where appropriate
+        if repetitions > 1:
+            names = ["simulation.%d" % num for num in range(repetitions)]
+        else:
+            names = "simulation"
+        return self.model._wrap_data(
+            y, start_idx, start_idx + nsimulations - 1, names=names
+        )
+
+    def forecast(self, steps=1, anchor=None):
```

Comment:
I think it would be better to follow the typical Statsmodels API in `tsa` for `predict` and `forecast`, which would be:

- `forecast` only performs out-of-sample forecasting that starts immediately after the sample ends.
- `predict` is the more general function that allows selecting both a `start` / `end`

So I think we should remove the `anchor` argument here (I know that this means a little refactoring, sorry about that!)

I'm not necessarily opposed to an `anchor` argument for `forecast` in general, but I think it would be better for us to consider whether we want to add it across the entire API rather than just here.

Reply:
Ah, yes, this was a point where I wasn't sure whether I understood it correctly. I now renamed my original `forecast` to `_forecast`, and am calling this function from the new `forecast`.